src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,195 @@
+                            System.out.println(Arrays.toString(event_str_arr));
+                            LocalDate from = LocalDate.parse(event_str_arr[1]);
+                            LocalDate to = LocalDate.parse(event_str_arr[2]);
+                            if (from.isAfter(to)){
+                                String err_msg = "☹ OOPS!!! Your time range is from a date to another date that is earlier" +
+                                        "than the former. Please key in a valid time range";
+                                throw new DukeException(err_msg);
+                            }
+                            Event event = new Event(event_str_arr[0].substring(6), from, to, 0);
+                            task_list.add(event);
+                            System.out.println("Got it. I've added this task");
+                            storage.update_data(task_list);
+
+                            System.out.println("added: " + event);
+                            System.out.println("Now you have " + task_list.size() + " tasks in the list");
+                        } catch (DateTimeParseException e) {
+                            String err_msg = "☹ OOPS!!! The description or date for the event is wrong, plase key in the" +
+                                    "date in the format of yyyy-mm-dd, eg. 2001-02-10\n"
+                                    + "You may key in: event hw1 /2001-02-10 /2001-02-12, Duke.Duke will record your event hw1 as" +
+                                    "from 2001-02-10 to 2001-02-12";
+                            throw new DukeException(err_msg);
+                        }
+                        continue;
+                    case "FIND":
+                        String[] find_input_arr = input.split(" ");
+                        if (task_list.size() == 0){
+                            String err_msg = "You have not upload any task yet";
+                            throw new DukeException(err_msg);
+                        }
+                        System.out.println("Here are the matching tasks in your list:");
+                        i = 1;
+                        for (Task task : task_list) {
+                            if (task.description.contains(find_input_arr[1])){
+                                System.out.println(i + ". " + task);
+                                i++;
+                            }
+                        }
+                        if(i == 1) {
+                            System.out.println("There are no matching task");
+                        }
+                        continue;
+                    case "ERROR":
+                        String err_msg = "☹ OOPS!!! I'm sorry, but I don't know what that means";
+                        throw new DukeException(err_msg);
+                    }
+                }catch (DukeException e){
+                    System.out.println(e);
+            }
+            input = ui.readCommand();
+        }
```

Comment:
According to the code quality standards, you need to [avoid long methods](https://nus-cs2103-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods). I suggest splitting the handling of commands into a separate function or class.